zpay32/invoice.go

```diff
@@ -849,6 +853,11 @@ func parseRoutingInfo(data []byte) ([]ExtraRoutingInfo, error) {
 		return nil, err
 	}
 
+	if len(base256Data)%routingInfoLen != 0 {
```

Comment:
I kept the check up there because we can avoid reading the data if it's not a multiple of the required length. For example, if we're reading two routing infos, but the second one is incomplete, we would avoid reading the first completely.